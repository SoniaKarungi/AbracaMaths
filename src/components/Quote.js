import React, { useState, useEffect } from 'react';

export default function Quote() {
  const [randomQuote, setRandomQuote] = useState('null');
  useEffect(() => {
    fetch('https://geek-quote-api.herokuapp.com/v1/quote')
      .then((response) => response.json())
      .then((data) => setRandomQuote(data));
  }, []);

  return (
    <div>
      <p
        data-testid="quote"
        style={{
          textAlign: 'center',
          fontSize: '20px',
          color: '#00ff15',
          background: 'black',
          padding: '20px',
        }}
      >
        {randomQuote.quote}
      </p>
      <p
        style={{
          color: 'grey',
          textAlign: 'center',
          fontSize: '20px',
        }}
      >
        {randomQuote.author}
      </p>
    </div>
  );
}
