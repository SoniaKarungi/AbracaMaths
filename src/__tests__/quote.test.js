import React from "react";
import renderer from "react-test-renderer";
import { render, screen, act } from "@testing-library/react";
import Quote from "../components/Quote";

describe("Quote", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
