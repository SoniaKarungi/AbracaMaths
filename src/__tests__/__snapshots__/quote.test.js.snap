// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote renders correctly 1`] = `
<div>
  <p
    data-testid="quote"
    style={
      Object {
        "background": "black",
        "color": "#00ff15",
        "fontSize": "20px",
        "padding": "20px",
        "textAlign": "center",
      }
    }
  />
  <p
    style={
      Object {
        "color": "grey",
        "fontSize": "20px",
        "textAlign": "center",
      }
    }
  />
</div>
`;
